s3/src/main/scala/org/apache/pekko/stream/connectors/s3/impl/S3Stream.scala

```diff
@@ -1177,11 +1177,15 @@ import scala.util.{ Failure, Success, Try }
 
         import conf.multipartUploadSettings.retrySettings._
 
-        SplitAfterSize(chunkSize, chunkBufferSize)(atLeastOneByteString)
-          .via(getChunkBuffer(chunkSize, chunkBufferSize, maxRetries)) // creates the chunks
-          .mergeSubstreamsWithParallelism(parallelism)
+        val source1: SubFlow[Chunk, NotUsed, Flow[ByteString, ByteString, NotUsed]#Repr, Sink[ByteString, NotUsed]] =
+          SplitAfterSize(chunkSize, chunkBufferSize)(atLeastOneByteString)
+            .via(getChunkBuffer(chunkSize, chunkBufferSize, maxRetries)) // creates the chunks
+
+        val source2 = source1.mergeSubstreamsWithParallelism(parallelism)
```

Comment:
The `source2` here seems to be a leftover from refactoring/figuring out types locally? You should be able to just go

```scala
source1.mergeSubstreamsWithParallelism(parallelism)
  .filter(_.size > 0)
  .via(atLeastOne)
  .zip(requestInfoOrUploadState(s3Location, contentType, s3Headers, initialUploadState))
  // etc etc
```

Reply:
the original code wouldn't compile in scala 3 and splitting the code up has helped to get it to compile

Reply:
code no longer compiles when I start removing the types on these params

Reply:
I can see that for `source1`, its `source2` that seems redundant (`source2` has no type annotation).